enh(mc): update infrastructure-active-directory-nrpe (#3200)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/infrastructure-active-directory-nrpe.md

---
id: infrastructure-active-directory-nrpe
title: Microsoft Active Directory NRPE
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> This Monitoring Connector monitors Active Directory Domain Controllers using local commands and the Centreon NSClient++ and the version 2 of the NRPE protocol agent to trigger their executions remotely.

Reviewer: ```suggestion
This Monitoring Connector monitors Active Directory Domain Controllers using local commands and the Centreon NSClient++ and version 2 of the NRPE protocol agent to trigger their executions remotely.
```

Reviewer: Tthere is an extra element compared to the French version. What are "local commands"?

Author: Commands that are executed locally on the windows server

## Pack assets

### Templates

The Monitoring Connector **Active Directory** brings a host template:

* **Infra-ActiveDirectory-NRPE-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="Infra-ActiveDirectory-NRPE-custom" label="Infra-ActiveDirectory-NRPE-custom">

| Service Alias        | Service Template                                    | Service Description                                                                   |
|:---------------------|:----------------------------------------------------|:--------------------------------------------------------------------------------------|

> | Ad-Domain-Controller | Infra-ActiveDirectory-Domain-Controller-NRPE-custom | This check makes to diagnose the domain controller. It executes the command "dcdiag"  |

Reviewer: ```suggestion
| Ad-Domain-Controller | Infra-ActiveDirectory-Domain-Controller-NRPE-custom | This check diagnoses the domain controller. It executes the "dcdiag" command |
```

> The services listed above are created automatically when the **Infra-ActiveDirectory-NRPE-custom** host template is used.

</TabItem>
<TabItem value="Not attached to a host template" label="Not attached to a host template">

| Service Alias | Service Template                               | Service Description |
|:--------------|:-----------------------------------------------|:--------------------|

> | Dfsr-Backlog  | Infra-ActiveDirectory-Dfsr-Backlog-NRPE-custom | Check dfsr backlog  |

Reviewer: ```suggestion
| Dfsr-Backlog  | Infra-ActiveDirectory-Dfsr-Backlog-NRPE-custom | Check the dfsr backlog  |
```

> The services listed above are not created automatically when a host template is applied. To use them, [create a service manually](/docs/monitoring/basic-objects/services), then apply the service template you want.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics linked to each service.

<Tabs groupId="sync">
<TabItem value="Ad-Domain-Controller" label="Ad-Domain-Controller">

> Coming soon

Reviewer: Coming soon!

> Coming soon

Author: ```suggestion
| Metric Name              | Unit  |
|:-------------------------|:------|
| domain controller status | N/A   |
```

| CONFIGFILE   | The command can be localized by using a configuration file. This parameter can be used to specify an alternative location for the configuration file.  |                   |             |
| EXTRAOPTIONS | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). |                   |             |

</TabItem>
<TabItem value="Dfsr-Backlog" label="Dfsr-Backlog">

| Macro            | Description                                                                                        | Default value     | Mandatory   |
|:-----------------|:---------------------------------------------------------------------------------------------------|:------------------|:-----------:|
| SENDINGMEMBER    | Name of the member that is sending the replication data. (mandatory)                               |                   | X           |
| REPLICATIONGROUP | Name for the replication group. (mandatory)                                                        |                   | X           |

> | REPLICATEDFOLDER | Name name for the replicated folder. (mandatory)                                                   |                   | X           |

Reviewer: name name

> | REPLICATIONGROUP | Name for the replication group. (mandatory)                                                        |                   | X           |
> | REPLICATEDFOLDER | Name name for the replicated folder. (mandatory)                                                   |                   | X           |

Author: ```suggestion
| REPLICATIONGROUP | Name of the replication group. (mandatory)                                                        |                   | X           |
| REPLICATEDFOLDER | Name of the replicated folder. (mandatory)                                                   |                   | X           |
```

```bash
/usr/lib64/nagios/plugins/check_nrpe -H 10.0.0.1 -p 5666 -t 30 -u -2 -P 8192 -c check_centreon_plugins -a 'apps::microsoft::activedirectory::local::plugin' 'dfsr-backlog'  ' \
	--sending-member="" \
	--replication-group="" \
	--replicated-folder="" \
	--warning-backlog="" \
	--critical-backlog="" '
```

The expected command output is shown below:

```bash
OK: Backlog File Count : 76 | 'backlog.file.count'=76;;;0;
```

### Troubleshooting

Please find the [troubleshooting documentation](../getting-started/how-to-guides/troubleshooting-plugins.md)
for Centreon Plugins typical issues.

### Available modes

In most cases, a mode corresponds to a service template. The mode appears in the execution command for the connector.
In the Centreon interface, you don't need to specify a mode explicitly: its use is implied when you apply a service template.
However, you will need to specify the correct mode for the template if you want to test the execution command for the 
connector in your terminal.

All available modes can be displayed by adding the `--list-mode` parameter to
the command:

```bash

> /usr/lib64/nagios/plugins/check_nrpe -H 10.0.0.1 -p 5666 -u -2 -P 8192 -t 30  -c check_centreon_plugins -a 'apps::microsoft::activedirectory::local::plugin' 'dfsr-backlog'  ' \

Reviewer: extra quote?

> /usr/lib64/nagios/plugins/check_nrpe -H 10.0.0.1 -p 5666 -u -2 -P 8192 -t 30  -c check_centreon_plugins -a 'apps::microsoft::activedirectory::local::plugin' 'dfsr-backlog'  ' \
> 	--sending-member="" \
> 	--list-mode

Author: ```suggestion
/usr/lib64/nagios/plugins/check_nrpe -H 10.0.0.1 -p 5666 -u -2 -P 8192 -t 30  -c check_centreon_plugins -a 'apps::microsoft::activedirectory::local::plugin' 'dfsr-backlog'  ' \
	--sending-member="" \
	--list-mode \
	'
```
